adapters/pubmatic/pubmatic.go

```diff
@@ -191,36 +239,34 @@ func parseImpressionObject(imp *openrtb2.Imp, wrapExt *string, pubID *string) er
 
 	var bidderExt ExtImpBidderPubmatic
 	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
-		return err
+		return wrapExt, pubID, err
 	}
 
 	var pubmaticExt openrtb_ext.ExtImpPubmatic
 	if err := json.Unmarshal(bidderExt.Bidder, &pubmaticExt); err != nil {
-		return err
+		return wrapExt, pubID, err
 	}
 
-	if *pubID == "" {
-		*pubID = strings.TrimSpace(pubmaticExt.PublisherId)
+	if extractPubIDFromImp {
+		pubID = strings.TrimSpace(pubmaticExt.PublisherId)
```

Comment:
We are passing boolean to indicate whether to extract publisher ID value from impression or not. This is just to avoid calling strings.TrimSpace function unnecessarily as parseImpressionObject function gets called for all impressions in the request and we need to extract publisher ID just once from any impression.
Earlier code was passing string pointer and setting value in the called function itself. @mansinahar pointed that it should be good to just extract value in the function and return it instead of setting value. So refactored the code to return publisher ID value.
Passing pubId pointer and updating value of it in parseImpressionObject function was misleading so refactored the code to extract and return publisher ID value on if necessary.